### DISCUSSION TITLE
Negative search filter

### DISCUSSION THREAD

**Discussion by The-furf-of-July** [modrinth/.github#262] [Feature requests]
While browsing mods I recently found myself wanting to exclude results from the search with the World Generation tag, due to wanting to continue using the same world.

Having the option to exclude specific tags while searching would be a good feature.
Selecting all the tags doesn't work, as that gets interpreted as requiring the mod to have every tag.

**Comment by The-furf-of-July** [comment]
Other potential use cases include excluding tech/magic mods to fit theming, or hiding "cursed" mods

**Comment by hiisuuii** [comment]
I think this would be a very good feature for all the checkboxes on the sidebar, including libraries and environments. Frequently I want to search the site for client-only or server-only mods, but I'm not able to do so, because there's no way to specify "yes client and not server" or "yes server and not client" as the environment, only "yes server and I don't care about client" or "yes client and I don't care about server", or "yes to both".

**Reply by Prospector** [reply]
I think you might misunderstand how the client and server filters work, they work as you would expect. Client for client-side mods, Server for server-side mods, neither selected for any, and both selected for client and serverside mods

**Reply by hiisuuii** [reply]
I think maybe I'm not communicating properly. Right now, if I tick the "server" box, it shows me any mods which have the "Server" environment supported, regardless of whether or not they also have the "Client" environment supported, for example Fabric API. What I'd like to be able to do is specifically search for server-only mods, which have no client-side component, like PolyFactory. I've attached some images so you can see what I mean.
This image is an example of what currently shows up when I select the "Server" environment tick-box:
![Screenshot_20240622_035724_Firefox.jpg](https://github.com/modrinth/.github/assets/17937620/b70b4751-93d8-4591-b95f-69a6a8256a46)
This is an example of the kind of content I'd like to be able to filter my search to. Notice how all of the mods only state they are for "Server" and not "Server and/or Client"
![Screenshot_20240622_040412_Firefox.jpg](https://github.com/modrinth/.github/assets/17937620/3af12dfe-dc8d-48aa-9288-281b559bece4)

**Reply by Prospector** [reply]
Interesting. Seems like an odd request that you would want to filter for a mod that works when only on a server, but only if it's incompatible with clients.

**Reply by hiisuuii** [reply]
Perhaps it is odd, but I still think it would be useful. Lately I've been searching for server-side content mods for my server, so I've had only "server" selected, but it's not quite as filtered as I'd like. A tri-state checkbox would be very useful since right now when searching, there's a lot of mods that get included which are actually primarily client mods, just with additional (optional) server-side features, like Jade, and a lot of "Inventory Tweaks" style mods.

**Reply by MadCow667** [reply]
I would really enjoy this feature.
You could click once to check the filter, making it inclusive, and then you could click it again to "x" the filter, causing it to be exclusive.

I'm looking for alternative font packs, and so many packs show up when only selecting "font" and "gui", it would be very nice to be able to exclude all the other categories

![image](https://github.com/modrinth/.github/assets/48391734/f6e65658-f3e1-4276-8685-8e6599446265)

**Comment by Minenash** [comment]
This has been added
